File: src/main/java/com/service/AccountService.java
package com.service;

import com.domain.DbAccount;
import com.repository.AccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.util.Date;


/**
 * class
 *
 * @author Five
 * @date create in 下午3:41 2017/11/30
 */
@Service
public class AccountService {

    @Autowired
    private AccountRepository accountRepository;

    /**
     * 获取账号信息
     *
     * @param account 账号
     * @return 账号信息
     */
    public DbAccount accountInfo(String account) {
        return accountRepository.findByAccount(account);
    }

    /**
     * 根据用户ID获取账号信息
     *
     * @param uid 用户id
     * @return
     */
    public DbAccount accountInfoByUid(String uid) {

        DbAccount rv = accountRepository.findByUid(uid);

        return rv;
    }

    /**
     * 注册账号
     *
     * @param account
     * @return
     */
    public DbAccount register(String account, String pssword, String uid) {
        DbAccount nv = new DbAccount();
        nv.setAccount(account);
        nv.setPassword(pssword);
        nv.setCreatetime(new Date());
        nv.setStatus(0);
        nv.setUid(uid);

        DbAccount rnv = accountRepository.save(nv);
        return rnv;
    }

    /**
     * 根据用户ID获取账号信息
     *
     * @param password 新密码
     * @param id       ID
     * @param dba      账号实体（可为空）
     * @return
     */
    @Transactional
    public Integer changePassword(String password, Integer id, DbAccount dba) {
        if (dba == null){
            return accountRepository.changedPassword(password, id);
        }
        else{
            dba.setPassword(password);
            dba.setId(id);

            accountRepository.save(dba);

            return 1;
        }
    }

}

File: src/main/java/com/service/AccountService.kt
package com.service

import com.domain.entity.DbAccount
import com.utils.ResponseResult

/**
 * class
 *
 * @author Five0314
 * @date create in 下午3:16 2017/12/14
 */
interface AccountService {

    /**
     * 登录
     *
     * @param account 账号
     * @param password 密码
     * @return
     */
    fun login(account: String, password: String) : ResponseResult<DbAccount>

    /**
     * 注册
     *
     * @param phone 手机号
     * @param password 密码
     * @return
     */
    fun register(phone: String, password: String) : ResponseResult<DbAccount>

    /**
     * 修改密码
     *
     * @param uid 用户编号
     * @param oldPassword 旧密码
     * @param newPassword 新密码
     * @return
     */
    fun changePassword(uid: String, oldPassword: String, newPassword: String) : ResponseResult<DbAccount>
}